src/core/reference/include/openvino/reference/function.hpp

```diff
@@ -13,5 +13,9 @@
 namespace ov {
 namespace reference {
 void function(const std::shared_ptr<Model>& function, const ov::TensorVector& inputs, ov::TensorVector& outputs);
```

Comment:
Is it possible to migrate `const EvaluationContext& evaluation_context` to original `function` as default arg?

Reply:
If there are two user can select explicitly to use context or not and there is no need to default initialization of not used context at least from reference impl perspective. However the Model's evaluate without context create context but is controlled in the Model. 
Have one function with default parameter will be also fine (as created context here will be always used by evaluate) but this default context will be created in two places instead of one.